syntax = "proto3";
package bank;

service Bank {
    rpc MsgDelivery (TransactionRequest) returns (TransactionResponse);
    rpc GetBranchEventsLog (BranchEventsLogRequest) returns (BranchEventsLogResponse);
}
message TransactionRequest {
    int32 customer_id = 1;
    int32 customer_request_id = 2;
    string operation = 3;
    int32 logical_clock = 4;
    int32 branch_id = 5;
    int32 amount = 6;
}
message TransactionResponse {
    string status = 1;
    int32 balance = 2;
}
message BranchEventsLogRequest {}
message BranchEventsLogResponse {
    repeated string events = 1;
}